### PR TITLE
use generic interface instead of accessing fields

### PR DESCRIPTION
**Pull request by SimonDanisch**
....since not all IOs have a lock field...

### CODE REVIEW

**Comment by ianshmean** [comment]
Seems like the right thing to do, but I'm not sure how to fix the tests like this that are breaking

https://github.com/JuliaIO/PNGFiles.jl/blob/249d996d9fd5ac841fd2914fb38dbd8741e3ca03/test/test_pngsuite.jl#L66

**Comment by SimonDanisch** [comment]
Fixed!

**Comment by ianshmean** [comment]
Awesome, thanks!

File: src/io.jl [hunk 1]
```diff
@@ -29,20 +29,24 @@ function load(fpath::String; gamma::Union{Nothing,Float64}=nothing, expand_palet
     close_png(fp)
     return out
 end
+
+maybe_lock(f, io::IO) = lock(f, io)
+# IOStream doesn't support locking...
+maybe_lock(f, io::IOStream) = f()
```
Comment (johnnychen94) [inline src/io.jl:35]: According to the report in https://github.com/JuliaImages/Images.jl/issues/912#issuecomment-690183454

It looks like `IOBuffer` should also be included as a special case?

Comment (SimonDanisch) [reply src/io.jl:35]: Probably? I guess you could add that to #21

**Pull request merged**